airflow/jobs.py

```diff
@@ -865,8 +865,8 @@ def create_dag_run(self, dag, session=None):
                     dag.start_date, next_run_date
                 )
 
-            # don't ever schedule in the future
-            if next_run_date > timezone.utcnow():
+            # don't ever schedule in the future or if next_run_date is None
+            if not next_run_date or next_run_date > timezone.utcnow():
```

Comment:
When is next run date None? Can we add a unit test for this?

Reply:
[task_start_dates](https://github.com/apache/airflow/blob/f3dfc9495be41221bd08ea753555df0975deb961/airflow/jobs.py#L840) list come out to be empty if there is no task available in dag's python file, and this cause  next_run_date be be None

I'll be adding UT(s) for this change soon.

Reply:
@ashb UT has been added, kindly have a look.

Reply:
@vardancse - next_run_date could be none for purely externally triggered DAGs? Wondering if this change caused https://jira.apache.org/jira/browse/AIRFLOW-4428

Reply:
@tooptoop4 AFAIK external triggered request never comes to this route, this is used for creation of scheduled dag_run. Fix was around graceful handling of date time exception, more details available here https://issues.apache.org/jira/browse/AIRFLOW-3962